### PR TITLE
RCInput: show source of RC input and rapidly switch between IOMCU and RCProt

### PR DESCRIPTION
**Pull request by tridge**
This changes the display of the RC input decoding to say:
```
2022-03-27 14:47:13.937 RCInput: decoding FPORT(2)
2022-03-27 14:47:49.451 RCInput: decoding SBUS(1)
```
the (n) is the source, where 1 is IOMCU, 2 is pulses on FMU and 3 is uart on FMU

This also fixes the failover between IOMCU and FMU inputs to be fast enough it won't trigger a failsafe on copter

I flew several flights on this today on a plane, with a IOMCU primary (RFD900x SBUS) and FPort secondary on telem2, on a CubeOrange

### CODE REVIEW

**Review by nkruzan — COMMENTED** [review]
minor nitpick

File: libraries/AP_HAL_ChibiOS/RCInput.cpp [hunk 1]
```diff
@@ -191,12 +192,13 @@ void RCInput::_timer_tick(void)
         _rx_link_quality = rcprot.get_rx_link_quality();
 #ifndef HAL_NO_UARTDRIVER
         rc_protocol = rcprot.protocol_name();
+        source = rcprot.using_uart()?RCSource::RCPROT_BYTES:RCSource::RCPROT_PULSES;
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/RCInput.cpp:195]: done

File: libraries/AP_HAL_ChibiOS/RCInput.h [hunk 2]
```diff
@@ -72,6 +72,15 @@ class ChibiOS::RCInput : public AP_HAL::RCInput {
     uint32_t _rcin_last_iomcu_ms;
     bool _init;
     const char *last_protocol;
+
+    enum class RCSource {
+        NONE = 0,
+        IOMCU = 1,
+        RCPROT_PULSES = 2,
+        RCPROT_BYTES = 3,
+        APRADIO = 4,
```
Comment (rmackay9) [inline libraries/AP_HAL_ChibiOS/RCInput.h:81]: AP_RADIO is a UART RC input I guess?

**Review by rmackay9 — APPROVED** [review]
I'm not totally sure that I'm qualified to review this but it seems ok to me.  At least I don't see any obvious errors.

File: libraries/AP_HAL_ChibiOS/RCInput.h (hunk 2 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/RCInput.h:81]: no, its SPI, used by skyviper

**Comment by tatsuy** [comment]
I tested this, and works well.
Thank you for this change!

**Pull request merged**